runtime/interpreter/interpreter_expression.go

```diff
@@ -382,6 +382,10 @@ func (interpreter *Interpreter) evalExpression(expression ast.Expression) Value
 }
 
 func (interpreter *Interpreter) checkInvalidatedResourceOrResourceReference(value Value, hasPosition ast.HasPosition) {
+	if interpreter == nil {
+		return
+	}
```

Comment:
Why would the receiver be `nil` here?

Reply:
There's a very specific possible sequence of calls that can get here: if `checkInvalidatedResourceOrResourceReference` is called from inside an array/dictionary/composite iteration (as is introduced in this PR), which is triggered by a call to `MeteredString`, which itself is triggered by a call to `RecursiveString` or `String`. While it makes sense to add the full interpreter as an argument to `MeteredString`, requiring the same for calls to `String` or `RecursiveString` is not reasonable, so they use a `nil` interpreter if they delegate to `MeteredString`.